src/api/java/com/minecolonies/api/crafting/IItemStorageFactory.java

```diff
@@ -32,9 +32,22 @@ default ItemStorage getNewInstance(@NotNull final IFactoryController factoryCont
             throw new IllegalArgumentException("Second parameter is supposed to be an Integer!");
         }
 
+        if (context.length >= PARAMS_ITEMSTORAGE + 1 && !(context[2] instanceof Boolean))
+        {
+            throw new IllegalArgumentException("Third parameter is supposed to be an Boolean!");
+        }
+
+        if (context.length >= PARAMS_ITEMSTORAGE + 2 && !(context[3] instanceof Boolean))
+        {
+            throw new IllegalArgumentException("Fourth parameter is supposed to be an Boolean!");
+        }
+
         final ItemStack stack = (ItemStack) context[0];
         final int size = (int) context[1];
-        return getNewInstance(stack, size);
+        final boolean ignoreDamage = context[2] != null ? (Boolean) context[2] : false;
+        final boolean ignoreNBT = context[3] != null ? (Boolean) context[3] : false;
```

Comment:
This needs to test `context.length` again instead of checking for null.

Reply:
Wow. Clearly distracted yesterday. Thanks.